### PR TITLE
Improve spec for Spree::Order#finalize!

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

I was looking at existing specs around stock allocation when an order is completed. And I stumbled across some ineffective specs of the Order model which I fixed here.

#### What should we test?

- Specs only.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/models/spree/order_spec.rb [hunk 1]
```diff
@@ -229,10 +229,12 @@
     end
 
     it "should sell inventory units" do
-      order.shipments.each do |shipment|
-        expect(shipment).to receive(:update!)
-        expect(shipment).to receive(:finalize!)
-      end
```
Comment (mkllnk) [inline spec/models/spree/order_spec.rb:230]: Sorry for the typo in the commit message. `order.shipments` was empty.

**Review by rioug — APPROVED** [review]
Nice find ! It's interesting to find out another way that spec can fail to test anything. Are shipments actually used in OFN ?

File: spec/models/spree/order_spec.rb [hunk 2]
```diff
@@ -220,8 +220,12 @@
...
 
     it "should set completed_at" do
-      expect(order).to receive(:touch).with(:completed_at)
-      order.finalize!
+      expect {
+        order.finalize!
+        order.reload
+      }.to change {
+        order.completed_at
+      }.from(nil)
```
Comment (dacook) [inline spec/models/spree/order_spec.rb:228]: Nice way to test this. I probably would have gone down the unnecessary task of freezing time and checking it changed `to` the current time..

**Review by dacook — APPROVED** [review]
Great improvements 💯

**Pull request merged**